lib/model/folder.go

```diff
@@ -164,52 +165,64 @@ func (f *folder) Serve(ctx context.Context) error {
 	initialCompleted := f.initialScanFinished
 
 	for {
+		var err error
+
 		select {
 		case <-f.ctx.Done():
 			close(f.done)
 			return nil
 
 		case <-f.pullScheduled:
-			f.pull()
+			_, err = f.pull()
 
 		case <-f.pullFailTimer.C:
-			if !f.pull() && f.pullPause < 60*f.pullBasePause() {
+			var success bool
+			success, err = f.pull()
+			if err == nil && success && f.pullPause < 60*f.pullBasePause() {
```

Comment:
This inverts the condition on `success` -- we shouldn't back off when pull was successfull